src/insexpand.c

```diff
@@ -2296,13 +2296,72 @@ set_ctrl_x_mode(int c)
     return retval;
 }
 
+    static char_u *
+ctrl_x_mode_to_string(int ctrl_x_mode)
+{
+    switch (ctrl_x_mode)
+    {
+	case CTRL_X_NORMAL:
+	    return (char_u *)"keyword";          // Default: CTRL-N, CTRL-P
+	case CTRL_X_NOT_DEFINED_YET:
+	    return (char_u *)"not-defined";     // Unused or undefined
+	case CTRL_X_SCROLL:
+	    return (char_u *)"scroll";           // For scroll completion
+	case CTRL_X_WHOLE_LINE:
+	    return (char_u *)"whole-line";       // Whole line completion
+	case CTRL_X_FILES:
+	    return (char_u *)"file";             // File name completion
+	case CTRL_X_TAGS:
+	    return (char_u *)"tag";              // Tag completion
+	case CTRL_X_PATH_PATTERNS:
+	    return (char_u *)"path-pattern";     // Included file keyword
+	case CTRL_X_PATH_DEFINES:
+	    return (char_u *)"path-define";      // Definitions/macros
+	case CTRL_X_DICTIONARY:
+	    return (char_u *)"dictionary";       // Dictionary completion
+	case CTRL_X_THESAURUS:
+	    return (char_u *)"thesaurus";        // Thesaurus completion
+	case CTRL_X_CMDLINE:
+	    return (char_u *)"cmdline";          // Vim command-line
+	case CTRL_X_FUNCTION:
+	    return (char_u *)"user";             // User-defined completion
+	case CTRL_X_OMNI:
+	    return (char_u *)"omni";             // Omni completion
+	case CTRL_X_SPELL:
+	    return (char_u *)"spell";            // Spelling suggestions
+	case CTRL_X_LOCAL_MSG:
+	    return (char_u *)"local-msg";        // Local message (internal)
+	case CTRL_X_EVAL:
+	    return (char_u *)"eval";             // Built-in function complete()
+	case CTRL_X_CMDLINE_CTRL_X:
+	    return (char_u *)"cmdline-ctrl-x";   // CTRL-X in command-line mode
+	default:
+	    return (char_u *)"unknown";          // Fallback for invalid mode
+    }
+}
```

Comment:
This is inconsistent with the values in `:h complete_info_mode`

Reply:
yep .. i think i can reuse them.